chore: support go generate and mod

justforlxz opened the pull request:
as title

Closed: https://github.com/linuxdeepin/developer-center/issues/955

justforlxz committed:
chore: support go generate and mod

as title

dengbo11 commented on gen.go:
```diff
@@ -0,0 +1,6 @@
+package gir
```
新增文件需要加上License Header， 下面文件同理

justforlxz committed:
chore: optimize directory structure

- rename gir-generator to generator
- move gi to generator/
- move go files from lib.in to generate dirs

justforlxz committed:
chore: update license

as title
